<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Gym</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="7,1">
    <item>
     <layout class="QVBoxLayout" name="scenePartLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="cameraBox">
        <property name="title">
         <string>Camera</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinAndMaxSize</enum>
         </property>
         <item row="0" column="0">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>56</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>&lt;center&gt;x&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraPosXBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>&lt;center&gt;y&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraPosYBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>&lt;center&gt;z&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraPosZBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>&lt;center&gt;position&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraPosXBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="cameraPosXBox">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="cameraPosYBox">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QDoubleSpinBox" name="cameraPosZBox">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>&lt;center&gt;center&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraCenterXBox</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="cameraCenterXBox">
           <property name="minimum">
            <double>-10.000000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="cameraCenterYBox">
           <property name="minimum">
            <double>-10.000000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QDoubleSpinBox" name="cameraCenterZBox">
           <property name="minimum">
            <double>-10.000000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>&lt;center&gt;up&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>cameraUpXBox</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="cameraUpXBox">
           <property name="minimum">
            <double>-2.000000000000000</double>
           </property>
           <property name="maximum">
            <double>2.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QDoubleSpinBox" name="cameraUpYBox">
           <property name="minimum">
            <double>-2.000000000000000</double>
           </property>
           <property name="maximum">
            <double>2.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QDoubleSpinBox" name="cameraUpZBox">
           <property name="minimum">
            <double>-2.000000000000000</double>
           </property>
           <property name="maximum">
            <double>2.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="perspectiveBox">
        <property name="title">
         <string>Perspective</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinAndMaxSize</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>&lt;center&gt;fovy&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>fovyBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="fovyBox">
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>179.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>&lt;center&gt;aspect&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>aspectBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QDoubleSpinBox" name="aspectBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>&lt;center&gt;near&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>nearBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="nearBox"/>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>&lt;center&gt;far&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>farBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QDoubleSpinBox" name="farBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="lightBox">
        <property name="title">
         <string>Light</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>&lt;center&gt;R&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightAmbientRBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>&lt;center&gt;G&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightAmbientGBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>&lt;center&gt;B&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightAmbientBBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>&lt;center&gt;A&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightAmbientABox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>&lt;center&gt;Ambient&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightAmbientRBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="lightAmbientRBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="lightAmbientGBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QDoubleSpinBox" name="lightAmbientBBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QDoubleSpinBox" name="lightAmbientABox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>&lt;center&gt;Diffuse&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lightDiffuseRBox</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="lightDiffuseRBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="lightDiffuseGBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QDoubleSpinBox" name="lightDiffuseBBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QDoubleSpinBox" name="lightDiffuseABox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_23">
           <property name="text">
            <string>&lt;center&gt;Ball&lt;/center&gt;</string>
           </property>
           <property name="buddy">
            <cstring>ballLightRBox</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="ballLightRBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QDoubleSpinBox" name="ballLightGBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QDoubleSpinBox" name="ballLightBBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="4">
          <widget class="QDoubleSpinBox" name="ballLightABox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="basketballBox">
        <property name="title">
         <string>Basket Ball</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QGroupBox" name="SphereBox">
             <property name="title">
              <string>Sphere</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>&lt;center&gt;radius&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>sphereRadiusBox</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="sphereRadiusBox">
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>&lt;center&gt;slices&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>sphereSlicesBox</cstring>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="sphereSlicesBox">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="singleStep">
                 <number>10</number>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>&lt;center&gt;stacks&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>sphereStacksBox</cstring>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="sphereStacksBox">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="singleStep">
                 <number>10</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="ballPosBox">
             <property name="title">
              <string>Position</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>&lt;center&gt;x&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>ballPosXBox</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="ballPosXBox">
                <property name="minimum">
                 <double>-10.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>&lt;center&gt;y&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>ballPosYBox</cstring>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="ballPosYBox">
                <property name="minimum">
                 <double>-10.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>&lt;center&gt;z&lt;/center&gt;</string>
                </property>
                <property name="buddy">
                 <cstring>ballPosZBox</cstring>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QDoubleSpinBox" name="ballPosZBox">
                <property name="minimum">
                 <double>-10.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
             <zorder>label_10</zorder>
             <zorder>label_11</zorder>
             <zorder>label_12</zorder>
             <zorder>ballPosXBox</zorder>
             <zorder>ballPosYBox</zorder>
             <zorder>ballPosZBox</zorder>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>cameraPosXBox</tabstop>
  <tabstop>cameraPosYBox</tabstop>
  <tabstop>cameraPosZBox</tabstop>
  <tabstop>cameraCenterXBox</tabstop>
  <tabstop>cameraCenterYBox</tabstop>
  <tabstop>cameraCenterZBox</tabstop>
  <tabstop>cameraUpXBox</tabstop>
  <tabstop>cameraUpYBox</tabstop>
  <tabstop>cameraUpZBox</tabstop>
  <tabstop>fovyBox</tabstop>
  <tabstop>aspectBox</tabstop>
  <tabstop>nearBox</tabstop>
  <tabstop>farBox</tabstop>
  <tabstop>lightAmbientRBox</tabstop>
  <tabstop>lightAmbientGBox</tabstop>
  <tabstop>lightAmbientBBox</tabstop>
  <tabstop>lightAmbientABox</tabstop>
  <tabstop>lightDiffuseRBox</tabstop>
  <tabstop>lightDiffuseGBox</tabstop>
  <tabstop>lightDiffuseBBox</tabstop>
  <tabstop>lightDiffuseABox</tabstop>
  <tabstop>ballLightRBox</tabstop>
  <tabstop>ballLightGBox</tabstop>
  <tabstop>ballLightBBox</tabstop>
  <tabstop>ballLightABox</tabstop>
  <tabstop>sphereRadiusBox</tabstop>
  <tabstop>sphereSlicesBox</tabstop>
  <tabstop>sphereStacksBox</tabstop>
  <tabstop>ballPosXBox</tabstop>
  <tabstop>ballPosYBox</tabstop>
  <tabstop>ballPosZBox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <slots>
  <signal>cameraPosChanged(GLdouble,GLdouble,GLdouble)</signal>
 </slots>
</ui>
